### PR TITLE
Switch to Jetbrains annotations

### PR DESCRIPTION
**Pull request by iGabyTM**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -51,6 +51,11 @@
     </repositories>
 
     <dependencies>
+        <dependency>
+            <groupId>org.jetbrains</groupId>
+            <artifactId>annotations</artifactId>
+            <version>20.1.0</version>
+        </dependency>
```
Comment (portlek) [inline pom.xml:58]: The score should be provided. The annotation classes are not effective within the classpath. It's for Intellij IDEA.

**Review by portlek — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (knightzmc) [reply pom.xml:58]: bruh what does that mean

Comment (iGabyTM) [reply pom.xml:58]: What do you mean? 🤔

Comment (portlek) [reply pom.xml:58]: scope*

Comment (portlek) [reply pom.xml:58]: I'm talking specifically about the annotations from Jetbrains, not other annotation libraries.

**Comment by CryptoMorin** [comment]
You provided no description why we should switch to IntelliJ annotations.

**Comment by portlek** [comment]
> You did not describe why we should switch to JetBrains annotations.

It's adding inspections for Intellij IDEA. This method or field can be null or not in compile time, not runtime.

**Comment by CryptoMorin** [comment]
The thing is, for some reasons IntelliJ highlights the warnings under the actual method parameter which shows you the usages with nulls. It doesn't highlight the warning on the usage itself. So it doesn't matter which nullability annotations you use. You can't see the warning if it's in the library. I don't even see a warning for it when I compile.
Or is there a new option added in IntelliJ that fixes this?

**Comment by portlek** [comment]
If you add your annotations in your dependencies(not from libraries) you can see.

**Comment by CryptoMorin** [comment]
I added JetBrains annotations to both XSeries and my other project libraries via Maven and it's still not showing the error. If it's only about showing which method returns null or not documentation-wise, it works for all annotations.

**Comment by iGabyTM** [comment]
As far as I know javax was removed from the jdk and Jetbrains' annotations is what most people use. Regarding the warnings, I see them on any project which has a library that uses the annotations.

**Comment by CryptoMorin** [comment]
Alright, my IntelliJ setting were old. I fixed them and now it's showing. I used both in XSeries and without needing to add them to my main project's classpath, it showed the warning just fine. It works with any nullability annotations.

**Pull request closed without merging**